### PR TITLE
Adds initial test coverage around the persistence of Lock names

### PR DESCRIPTION
**Pull request by akeem**
# Description

Expanding on the work to provide better integration test coverage for the application, this PR adds  coverage around a name being saved.

# Issues

Fixes #
Refs #1907

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
Dang. any idea why the tests fail? Could it be that the samrt contract is still not deployed where we expected it to be?

**Comment by akeem** [comment]
@julien51 this is what I was working on this morning. It was/is a configuration issue the contract is there but this is around locksmith

**Comment by akeem** [comment]
I've updated out CI environments include values for the following environment variables:

* CI
* HTTP_PROVIDER
* DB_USERNAME
* DB_PASSWORD
* DB_NAME
* DB_HOSTNAME
* LOCKSMITH_URI

I do believe that our integration test script may not be functioning exactly as we believe it may. https://forums.docker.com/t/environment-variables-in-args/27522/2

**Comment by cellog** [comment]
I wonder... docker created swarms to support what docker-compose does, and has been pushing folks to use that instead of docker-compose. Perhaps we should look at whether that would make things more reliable since it's built into docker?

https://vsupalov.com/difference-docker-compose-and-docker-stack/

**Comment by julien51** [comment]
Let's keep things simple and avoid yet another rabbit hole. Docker compose should work fine for now ;)

File: docker/docker-compose.ci.yml [hunk 1]
```diff
@@ -53,17 +56,20 @@ services:
     depends_on:
       - ganache-integration
       - unlock
+      - locksmith
```
Comment (julien51) [inline docker/docker-compose.ci.yml]: I think we should not have it both here and in `unlock` above. I'm in favor of just having it here but that's not a huge deal either way.

File: docker/docker-compose.ci.yml [hunk 2]
```diff
@@ -53,17 +56,20 @@ services:
...
     working_dir: /home/unlock/tests
   locksmith:
     environment:
       - DB_USERNAME
       - DB_PASSWORD
       - DB_NAME
       - DB_HOSTNAME
+      - NODE_ENV=production
```
Comment (julien51) [inline docker/docker-compose.ci.yml]: Hum. Is that required?

File: docker/docker-compose.ci.yml [hunk 3]
```diff
@@ -53,17 +56,20 @@ services:
...
     image: unlock
     user: node
     working_dir: /home/unlock/locksmith
-    ports:
+    command: sh -c './scripts/wait-for.sh db:5432 -- npm run db:migrate && npm run start'
```
Comment (julien51) [inline docker/docker-compose.ci.yml]: Yikes.

File: unlock-app/src/config.js [hunk 4]
```diff
@@ -97,7 +97,7 @@ export default function configure(
     )
     blockTime = 10 // in mseconds.
     supportedProviders = ['HTTP']
-    services['storage'] = { host: 'http://127.0.0.1:8080' }
+    services['storage'] = { host: runtimeConfig.locksmithHost }
```
Comment (julien51) [inline unlock-app/src/config.js]: Is there a default value for this set in `config.next.js`? I just want to make sure things run fine in dev...

File: scripts/integration-tests.sh [hunk 5]
```diff
@@ -1,9 +1,20 @@
 #!/usr/bin/env bash
 
+
```
Comment (julien51) [inline scripts/integration-tests.sh]: nit: newline?

File: scripts/integration-tests.sh [hunk 6]
```diff
@@ -1,9 +1,20 @@
...
+export DB_USERNAME='locksmith_test'
+export DB_PASSWORD='password'
+export DB_NAME='locksmith_test'
+export DB_HOSTNAME='db'
+export CI=true
+export HTTP_PROVIDER='ganache-integration'
+export LOCKSMITH_URI='http://locksmith:8080'
+
+
```
Comment (julien51) [inline scripts/integration-tests.sh]: nit: same same ;)

File: docker/docker-compose.ci.yml (hunk 2 above)
Comment (akeem) [reply docker/docker-compose.ci.yml]: We could probably come up with another environment for this, but the non-production environment will pull in the dev packages and will bring sqllite along with it.

**Pull request merged**